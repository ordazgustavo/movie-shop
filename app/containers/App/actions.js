/*
 * App Actions
 *
 * Actions change things in your application
 * Since this boilerplate uses a uni-directional data flow, specifically redux,
 * we have these actions which are the only way your application interacts with
 * your application state. This guarantees that your state is up to date and nobody
 * messes it up weirdly somewhere.
 *
 * To add a new Action:
 * 1) Import your constant
 * 2) Add a function like this:
 *    export function yourAction(var) {
 *        return { type: YOUR_ACTION_CONSTANT, var: var }
 *    }
 */

import {
  GET_POPULAR_MOVIES_REQUEST,
  GET_POPULAR_MOVIES_SUCCESS,
  GET_POPULAR_MOVIES_FAILURE,
  GET_TOP_RATED_MOVIES_REQUEST,
  GET_TOP_RATED_MOVIES_SUCCESS,
  GET_TOP_RATED_MOVIES_FAILURE,
  GET_UPCOMING_MOVIES_REQUEST,
  GET_UPCOMING_MOVIES_SUCCESS,
  GET_UPCOMING_MOVIES_FAILURE,
} from './constants'

export function getPopularMoviesRequest() {
  return {
    type: GET_POPULAR_MOVIES_REQUEST,
  }
}

export function getPopularMoviesSuccess(popularMovies) {
  return {
    type: GET_POPULAR_MOVIES_SUCCESS,
    popularMovies,
  }
}

export function getPopularMoviesFailure(error) {
  return {
    type: GET_POPULAR_MOVIES_FAILURE,
    error,
  }
}

export function getTopRatedMoviesRequest() {
  return {
    type: GET_TOP_RATED_MOVIES_REQUEST,
  }
}

export function getTopRatedMoviesSuccess(topRatedMovies) {
  return {
    type: GET_TOP_RATED_MOVIES_SUCCESS,
    topRatedMovies,
  }
}

export function getTopRatedMoviesFailure(error) {
  return {
    type: GET_TOP_RATED_MOVIES_FAILURE,
    error,
  }
}

export function getUpcomingMoviesRequest() {
  return {
    type: GET_UPCOMING_MOVIES_REQUEST,
  }
}

export function getUpcomingMoviesSuccess(upcomingMovies) {
  return {
    type: GET_UPCOMING_MOVIES_SUCCESS,
    upcomingMovies,
  }
}

export function getUpcomingMoviesFailure(error) {
  return {
    type: GET_UPCOMING_MOVIES_FAILURE,
    error,
  }
}
