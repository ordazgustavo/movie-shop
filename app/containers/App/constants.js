/*
 * AppConstants
 * Each action has a corresponding type, which the reducer knows and picks up on.
 * To avoid weird typos between the reducer and the actions, we save them as
 * constants here. We prefix them with 'yourproject/YourComponent' so we avoid
 * reducers accidentally picking up actions they shouldn't.
 *
 * Follow this format:
 * export const YOUR_ACTION_CONSTANT = 'yourproject/YourContainer/YOUR_ACTION_CONSTANT';
 */

export const GET_POPULAR_MOVIES_REQUEST = 'shop/App/GET_POPULAR_MOVIES_REQUEST'
export const GET_POPULAR_MOVIES_SUCCESS = 'shop/App/GET_POPULAR_MOVIES_SUCCESS'
export const GET_POPULAR_MOVIES_FAILURE = 'shop/App/GET_POPULAR_MOVIES_FAILURE'

export const GET_TOP_RATED_MOVIES_REQUEST =
  'shop/App/GET_TOP_RATED_MOVIES_REQUEST'
export const GET_TOP_RATED_MOVIES_SUCCESS =
  'shop/App/GET_TOP_RATED_MOVIES_SUCCESS'
export const GET_TOP_RATED_MOVIES_FAILURE =
  'shop/App/GET_TOP_RATED_MOVIES_FAILURE'

export const GET_UPCOMING_MOVIES_REQUEST =
  'shop/App/GET_UPCOMING_MOVIES_REQUEST'
export const GET_UPCOMING_MOVIES_SUCCESS =
  'shop/App/GET_UPCOMING_MOVIES_SUCCESS'
export const GET_UPCOMING_MOVIES_FAILURE =
  'shop/App/GET_UPCOMING_MOVIES_FAILURE'
