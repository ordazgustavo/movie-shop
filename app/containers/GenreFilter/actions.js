/*
 *
 * GenreFilter actions
 *
 */

import {
  GET_GENRES_REQUEST,
  GET_GENRES_SUCCESS,
  GET_GENRES_FAILURE,
  CHANGE_GENRE,
} from './constants'

export function getGenresRequest() {
  return {
    type: GET_GENRES_REQUEST,
  }
}

export function getGenresSuccess(genres) {
  return {
    type: GET_GENRES_SUCCESS,
    genres,
  }
}

export function getGenresFailure(error) {
  return {
    type: GET_GENRES_FAILURE,
    error,
  }
}

export function changeGenre(genreId) {
  return {
    type: CHANGE_GENRE,
    genreId,
  }
}
