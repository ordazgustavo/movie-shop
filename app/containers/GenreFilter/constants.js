/*
 *
 * GenreFilter constants
 *
 */

export const GET_GENRES_REQUEST = 'shop/GenreFilter/GET_GENRES_REQUEST'
export const GET_GENRES_SUCCESS = 'shop/GenreFilter/GET_GENRES_SUCCESS'
export const GET_GENRES_FAILURE = 'shop/GenreFilter/GET_GENRES_FAILURE'

export const CHANGE_GENRE = 'shop/GenreFilter/CHANGE_GENRE'
