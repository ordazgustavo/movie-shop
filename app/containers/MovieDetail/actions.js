/*
 *
 * MovieDetail actions
 *
 */

import {
  GET_MOVIE_DETAIL_REQUEST,
  GET_MOVIE_DETAIL_SUCCESS,
  GET_MOVIE_DETAIL_FAILURE,
} from './constants'

export function getMovieDetailRequest(movieId) {
  return {
    type: GET_MOVIE_DETAIL_REQUEST,
    movieId,
  }
}

export function getMovieDetailSuccess(movieDetail) {
  return {
    type: GET_MOVIE_DETAIL_SUCCESS,
    movieDetail,
  }
}

export function getMovieDetailFailure(error) {
  return {
    type: GET_MOVIE_DETAIL_FAILURE,
    error,
  }
}
