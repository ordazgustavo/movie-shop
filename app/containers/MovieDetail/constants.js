/*
 *
 * MovieDetail constants
 *
 */

export const GET_MOVIE_DETAIL_REQUEST =
  'shop/MovieDetail/GET_MOVIE_DETAIL_REQUEST'
export const GET_MOVIE_DETAIL_SUCCESS =
  'shop/MovieDetail/GET_MOVIE_DETAIL_SUCCESS'
export const GET_MOVIE_DETAIL_FAILURE =
  'shop/MovieDetail/GET_MOVIE_DETAIL_FAILURE'
