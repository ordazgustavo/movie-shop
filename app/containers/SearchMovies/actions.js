/*
 *
 * SearchMovies actions
 *
 */

import {
  SET_SEARCH_QUERY,
  SEARCH_MOVIES_REQUEST,
  SEARCH_MOVIES_SUCCESS,
  SEARCH_MOVIES_FAILURE,
} from './constants'

export function setSearchQuery(query) {
  return {
    type: SET_SEARCH_QUERY,
    query,
  }
}

export function searchMoviesRequest() {
  return {
    type: SEARCH_MOVIES_REQUEST,
  }
}

export function searchMoviesSuccess(movies) {
  return {
    type: SEARCH_MOVIES_SUCCESS,
    movies,
  }
}

export function searchMoviesFailure(error) {
  return {
    type: SEARCH_MOVIES_FAILURE,
    error,
  }
}
