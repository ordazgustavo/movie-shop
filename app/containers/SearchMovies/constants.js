/*
 *
 * SearchMovies constants
 *
 */

export const SET_SEARCH_QUERY = 'shop/SearchMovies/SET_SEARCH_QUERY'

export const SEARCH_MOVIES_REQUEST = 'shop/SearchMovies/SEARCH_MOVIES_REQUEST'
export const SEARCH_MOVIES_SUCCESS = 'shop/SearchMovies/SEARCH_MOVIES_SUCCESS'
export const SEARCH_MOVIES_FAILURE = 'shop/SearchMovies/SEARCH_MOVIES_FAILURE'
