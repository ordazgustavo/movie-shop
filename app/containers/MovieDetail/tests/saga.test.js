/**
 * Test sagas
 */

/* eslint-disable redux-saga/yield-effects */
import { put, takeLatest } from 'redux-saga/effects'

import { GET_MOVIE_DETAIL_REQUEST } from '../constants'
import { getMovieDetailSuccess, getMovieDetailFailure } from '../actions'

import movieDetailSaga, { getMovieDetail } from '../saga'

const movieId = 123

describe('getMovieDetail Saga', () => {
  let getMovieDetailGenerator

  // We have to test twice, once for a successful load and once for an unsuccessful one
  // so we do all the stuff that happens beforehand automatically in the beforeEach
  beforeEach(() => {
    getMovieDetailGenerator = getMovieDetail()

    const selectDescriptor = getMovieDetailGenerator.next().value
    expect(selectDescriptor).toMatchSnapshot()

    const callDescriptor = getMovieDetailGenerator.next(movieId).value
    expect(callDescriptor).toMatchSnapshot()
  })

  it('should dispatch the getMovieDetailSuccess action if it requests the data successfully', () => {
    const response = {
      id: 123,
      title: 'Avengers',
    }
    const putDescriptor = getMovieDetailGenerator.next(response).value
    expect(putDescriptor).toEqual(put(getMovieDetailSuccess(response)))
  })

  it('should call the getMovieDetailFailure action if the response errors', () => {
    const response = new Error('Some error')
    const putDescriptor = getMovieDetailGenerator.throw(response).value
    expect(putDescriptor).toEqual(put(getMovieDetailFailure(response)))
  })
})

describe('movieDetailSaga Saga', () => {
  const githubDataSaga = movieDetailSaga()

  it('should start task to watch for SET_SEARCH_QUERY action', () => {
    const takeLatestDescriptor = githubDataSaga.next().value
    expect(takeLatestDescriptor).toEqual(
      takeLatest(GET_MOVIE_DETAIL_REQUEST, getMovieDetail),
    )
  })
})
