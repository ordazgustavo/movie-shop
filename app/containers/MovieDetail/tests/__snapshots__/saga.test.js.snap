// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getMovieDetail Saga should call the getMovieDetailFailure action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`getMovieDetail Saga should call the getMovieDetailFailure action if the response errors 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "/movie/123?language=en-US&page=1",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`getMovieDetail Saga should dispatch the getMovieDetailSuccess action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`getMovieDetail Saga should dispatch the getMovieDetailSuccess action if it requests the data successfully 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "/movie/123?language=en-US&page=1",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;
